src/sqlfluff/rules/L049.py

```diff
@@ -0,0 +1,65 @@
+"""Implementation of Rule L006."""
+
+
+from sqlfluff.core.rules.base import BaseRule, LintResult
+from sqlfluff.core.rules.doc_decorators import document_fix_compatible
+
+
+@document_fix_compatible
+class Rule_L049(BaseRule):
+    """Comparisons with NULL should use "IS" or "IS NOT".
+
+    | **Anti-pattern**
+    | In this example, the <> is used to check for NULL values'.
+
+    .. code-block:: sql
+
+        SELECT
+            a
+        FROM foo
+        WHERE a <> NULL
+
+
+    | **Best practice**
+    | Use "IS" or "IS NOT" to check for NULL values.
+
+    .. code-block:: sql
+
+        SELECT
+            a
+        FROM foo
+        WHERE a IS NOT NULL
+    """
+
+    def _eval(self, segment, **kwargs):
+        """Relational operators should not be used to check for NULL values."""
+        # Iterate through children of this segment looking for equals or "not
+        # equals".
+        if len(segment.segments) <= 2:
+            return LintResult()
+
+        operator = None
+        for idx, sub_seg in enumerate(segment.segments):
+            # Skip anything which is whitespace or non-code.
+            if sub_seg.is_whitespace or not sub_seg.is_code:
+                continue
+            # Look for "=" or "<>".
```

Comment:
What about `>` or `<`? Do they work with `NULL`?

Should you check for `IS` and `NOT NULL` and complain on anything else?

Reply:
SQLFluff can parse `SELECT 1 > NULL`, but BigQuery reports a syntax error: `Operands of > cannot be literal NULL`. Not sure about other databases.

I'm torn about whether to make this change. It's much more likely `=` and `<>` would be used unknowingly, because it seems to make sense. Unless you or someone else feels strongly about changing it, I would prefer (for now at least) to keep it "as is".